br/pkg/lightning/mydump/csv_parser.go

```diff
@@ -175,18 +194,30 @@ func encodeSpecialSymbols(cfg *config.CSVConfig, cc *CharsetConvertor) (separato
 	return
 }
 
-func (parser *CSVParser) unescapeString(input string) (unescaped string, isNull bool, err error) {
+func (parser *CSVParser) unescapeString(input field) (unescaped string, isNull bool, err error) {
 	// Convert the input from another charset to utf8mb4 before we return the string.
-	if input, err = parser.charsetConvertor.Decode(input); err != nil {
+	if unescaped, err = parser.charsetConvertor.Decode(input.content); err != nil {
 		return
 	}
-	if parser.escFlavor == backslashEscapeFlavorMySQLWithNull && input == `\N` {
-		return input, true, nil
+	if parser.escFlavor == escapeFlavorMySQLWithNull && unescaped == parser.escapedBy+`N` {
+		return input.content, true, nil
+	}
+	if len(parser.escapedBy) > 0 {
+		unescaped = unescape(unescaped, "", parser.escFlavor, parser.escapedBy[0], parser.unescapeRegexp)
+	}
+	if len(parser.quote) == 0 || !parser.quotedNullIsText {
+		isNull = parser.escFlavor != escapeFlavorMySQLWithNull &&
+			!parser.cfg.NotNull &&
+			slices.Contains(parser.cfg.Null, unescaped)
```

Comment:
`unescaped == parser.cfg.Null` ? 🤔

Reply:
Sorry it must be a slice, because lightning will not always treat `\N` as NULL
https://docs.pingcap.com/zh/tidb/stable/tidb-lightning-data-source#backslash-escape

So at least for LOAD DATA under MySQL semantics, I need to set `["NULL", "\N"]`

Reply:
Oh, I thought null was a string.